### PR TITLE
Bump TypeScript to 4.8.4 and remove ttypescript 🉐

### PR DESCRIPTION
**Pull request by adriencaccia**
## TypeScript bump
Bump the whole repo to TypeScript `4.8.4`.

## TypeScript absolute paths mappings in packages `d.ts` files
Remove `ttypescript` and [`@zerollup/ts-transform-paths`](https://github.com/zerkalica/zerollup/tree/master/packages/ts-transform-paths) in favor of [`tsc-alias`](https://github.com/justkey007/tsc-alias).

### Reasons for the change
- No need to use `ttypescript` anymore
- We can trade two dependencies for a single one
- `tsc-alias` is actively maintained, whereas `@zerollup/ts-transform-paths` is not

### CODE REVIEW

**Comment by adriencaccia** [comment]
@Sc0ra @fargito
WDYT?

File: packages/nx-plugin/src/generators/library/typed-json-config/tsconfig.json.ts [hunk 1]
```diff
@@ -7,8 +7,6 @@ export const packageTsConfig = (options: NormalizedSchema): TsConfig => ({
   compilerOptions: {
     baseUrl: 'src',
```
Comment (fargito) [inline packages/nx-plugin/src/generators/library/typed-json-config/tsconfig.json.ts:8]: Does tsconfig.build.json need to be updated as well?

Comment (adriencaccia) [reply packages/nx-plugin/src/generators/library/typed-json-config/tsconfig.json.ts:8]: Yes, nicely spotted 👀

**Review by fargito — APPROVED** [review]
LGTM!

**Pull request merged**